plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/api/color/ColorManager.java

```diff
@@ -144,4 +163,127 @@ public Map<String, RGB> collectVsmAndDefaultColors(Session session) {
+     * 
+     * @param session
+     *            the session from which the VSMs defining colors can be retrieved.
+     * @return the map of RGB colors and their name defined in VSM
+     */
+    public Map<String, RGB> collectVsmColors(Session session) {
+        Map<String, RGB> vsmColors = new LinkedHashMap<>();
+        final IPreferenceStore preferenceStore = SiriusEditPlugin.getPlugin().getPreferenceStore();
+        if (preferenceStore != null) {
+            List<UserColorsPalette> palettes = session.getSelectedViewpoints(true).stream().map(EObject::eContainer).map(org.eclipse.sirius.viewpoint.description.Group.class::cast)
+                    .flatMap(group -> group.getUserColorsPalettes().stream()).collect(Collectors.toList());
+
+            vsmColors = palettes.stream().flatMap(palette -> palette.getEntries().stream()).filter(UserFixedColor.class::isInstance).map(UserFixedColor.class::cast)
+                    .collect(Collectors.toMap(c -> c.getName(), c -> new RGB(c.getRed(), c.getGreen(), c.getBlue()), (rgb1, rgb2) -> {
+                        // in a case of collision of key we keep the first value
+                        return rgb1;
+                    }, () -> new LinkedHashMap<String, RGB>()));
+        }
+        return vsmColors;
+    }
+
+    /**
+     * Gets the last colors used by the user in a color palette.
+     * 
+     * @return the list of last used colors.
+     */
+    public List<RGB> getLastUsedColors() {
+        String lastUsedColors = DiagramUIPlugin.getPlugin().getPreferenceStore().getString(SiriusDiagramUiPreferencesKeys.PREF_LAST_USED_COLORS.name());
+        List<RGB> colorsList = new ArrayList<>();
+        if (!lastUsedColors.isEmpty()) {
+            for (String s : lastUsedColors.split("},")) { //$NON-NLS-1$
+                RGB color = stringToRGB(s);
+                colorsList.add(color);
+            }
+        }
+        return colorsList;
+    }
+
+    /**
+     * Adds a new color to the {@link SiriusDiagramUiPreferencesKeys.PREF_LAST_USED_COLORS} preference.
+     * 
+     * @param newColor
+     *            the color to add.
+     */
+    public void addLastUsedColor(RGB newColor) {
+        List<RGB> lastUsedColors = getLastUsedColors();
+        if (newColor != null) {
+            if (!lastUsedColors.contains(newColor)) {
+                lastUsedColors.add(0, newColor);
+            } else {
+                lastUsedColors.remove(newColor);
+                lastUsedColors.add(0, newColor);
+            }
+            String lastUsedColorsString = new String();
+            for (int i = 0; i < lastUsedColors.size() && i < NB_MAX_LAST_USED_COLORS; i++) {
+                RGB rgb = lastUsedColors.get(i);
+                lastUsedColorsString = lastUsedColorsString + rgbToString(rgb) + RGB_VALUES_SEPARATOR;
+            }
+            IEclipsePreferences diagramUiPreferences = DefaultScope.INSTANCE.getNode(DiagramUIPlugin.ID);
```

Comment:
use the same preferenceStore as the one you are getting the value.
DiagramUIPlugin.getPlugin().getPreferenceStore()